core/tests/test_utils.py

```diff
@@ -2282,6 +2280,118 @@ def assert_same_list_elements(self, phrases, stdout):
+            html_body, bcc=None, reply_to=None, recipient_variables=None):
+        """Mocks sending an email to each email in recipient_emails.
+
+        Args:
+            sender_email: str. The email address of the sender. This should be
+                in the form 'SENDER_NAME <SENDER_EMAIL_ADDRESS>' or
+                'SENDER_EMAIL_ADDRESS'. Must be utf-8.
+            recipient_emails: list(str). The email addresses of the recipients.
+                Must be utf-8.
+            subject: str. The subject line of the email, Must be utf-8.
+            plaintext_body: str. The plaintext body of the email. Must be utf-8
+            html_body: str. The HTML body of the email. Must fit in a datastore
+                entity. Must be utf-8.
+            bcc: list(str)|None. List of bcc emails. Must be utf-8.
+            reply_to: str|None. Reply address formatted like
+                “reply+<reply_id>@<incoming_email_domain_name>
+                reply_id is the unique id of the sender.
+            recipient_variables: dict|None. If batch sending requires
+                differentiating each email based on the recipient, we assign a
+                unique id to each recipient, including info relevant to that
+                recipient so that we can reference it when composing the email
+                like so:
+                    recipient_variables =
+                        {"bob@example.com": {"first":"Bob", "id":1},
+                        "alice@example.com": {"first":"Alice", "id":2}}
+                    subject = 'Hey, %recipient.first%’
+                More info about this format at:
+                https://documentation.mailgun.com/en/
+                    latest/user_manual.html#batch-sending
+
+        Returns:
+            bool. Whether the email is sent succesfully.
+        """
+        bcc_emails = None
+
+        if bcc:
+            bcc_emails = bcc[0] if len(bcc) == 1 else bcc
+
+        new_email = self.EmailMessageMock(
+            sender_email, recipient_emails, subject, plaintext_body, html_body,
+            bcc=bcc_emails, reply_to=(reply_to if reply_to else None),
+            recipient_variables=(
+                recipient_variables if (recipient_variables) else None))
+        for recipient_email in recipient_emails:
+            if recipient_email not in self.emails_dict:
+                self.emails_dict[recipient_email] = []
+            self.emails_dict[recipient_email].append(new_email)
+        return True
+
+    def _get_sent_email_messages(self, to):
+        """Gets messages to a single recipient email.
+
+        Args:
+            to: str. The recipient email address.
+
+        Returns:
+            list(EmailMessageMock). Returns the list of email messages
+            corresponding to that recipient email.
+        """
+        return self.emails_dict[to] if to in self.emails_dict else []
+
+    def _get_all_sent_email_messages(self):
+        """Gets the entire messages dictionary."""
```

Comment:
Missing returns docstring. (Make sure it's clear about the entire type specification of what's being returned.)

Reply:
Done